wrappers/csharp/src/Device.cs

```diff
@@ -154,5 +153,52 @@ public SensorList Sensors
             }
         }
     }
+	
+	public class AdvancedDevice : Device
+    {
+        private const string AdvancedModeDisabledException = "Advanced mode has not been enabled";
+
+        internal AdvancedDevice(IntPtr dev) : base(dev)
+        {
+
+        }
+
+        public bool AdvancedMode
```

Comment:
Now that it is inside the class, I think it will be clearer to call this `Enabled` or something